LayoutTests/imported/w3c/web-platform-tests/html/rendering/widgets/input-text-size.html

```diff
@@ -0,0 +1,45 @@
+<!DOCTYPE html>
+<link rel="help" href="https://html.spec.whatwg.org/C/#the-input-element-as-a-text-entry-widget">
+<title>Test `size` attribute behaivor</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+
+<input id="missing">
+<input id="invalid" size="-1">
+<input id="size0" size="0">
+<input id="size1" size="1">
+<input id="size10" size="10">
+<input id="size20" size="20">
+<input id="size21" size="21">
+<input id="computed" style="border:none; padding:0;" size="19">
+<input id="computedNone" style="display:none" size="17">
+
+<script>
+test(() => {
+  assert_equals(missing.offsetWidth, size20.offsetWidth);
+}, 'A misssing attribute is equivalent to size=20');
+
+test(() => {
+  assert_equals(invalid.offsetWidth, size20.offsetWidth, 'size="-1"');
+  assert_equals(size0.offsetWidth, size20.offsetWidth, 'size="0"');
+}, 'An invalid attribute value is equivalent to size=20');
+
+test(() => {
+  assert_less_than(size1.offsetWidth, size10.offsetWidth, '1 < 10');
+  assert_less_than(size10.offsetWidth, size20.offsetWidth, '10 < 20');
+  assert_less_than(size20.offsetWidth, size21.offsetWidth, '20 < 21');
+}, 'The width depends on a size attribute');
+
+test(() => {
+  const computedString = getComputedStyle(computed).width;
+  assert_equals(computed.offsetWidth,
+      Math.round(computedString.substring(0, computedString.length - 2)));
```

Comment:
Change from parseInt to Math.round is upstreamed at https://github.com/web-platform-tests/wpt/pull/46403